result_report.py

```diff
@@ -1,31 +1,66 @@
 from robot.api import ExecutionResult, ResultVisitor
 import sys
+import requests
 
 class ResultReport(ResultVisitor):
     def __init__(self, markdown_file='report.md'):
-        self.failed_tests = []
-        self.passed_tests = []
+        self.failed_tests = 0
+        self.passed_tests = 0
+        self.tests = []
         self.markdown_file = markdown_file
 
     def visit_test(self, test):
         if test.status == 'FAIL':
-            self.failed_tests.append(test.name)
+            self.failed_tests += 1
         elif test.status == 'PASS':
-            self.passed_tests.append(test.name)
+            self.passed_tests += 1
+        self.tests.append(test)
+
+    def add_component_version_table(self, file):
+        if len(endpoints) == 0:
+            return None
+        
+        endpoint_versions = {}
+
+        urls = endpoints.split(',')
+        for url in urls:
+            if len(username) > 0:
+                response = requests.get(url + '/version.json', auth=(username, password))
+            else:
+                response = requests.get(url + '/version.json')
+
+            if response.status_code == 200:
+                version = response.json()['version'] 
+            else:
+                version = '**ERROR ' + str(response.status_code) + '**'
+
+            endpoint_versions[url.replace('https://','')] = version
+        
+        file.write("Tested components:\n")
+        file.write("| Component | Version |\n")
+        file.write("| -- | -- |\n")
+        for endpoint, version in endpoint_versions.items():
+            file.write("| " + endpoint + " | " + version + " |\n")
+        file.write("\n")
+        return None
 
     def end_result(self, result):
         # Create a new markdown file
         with open(self.markdown_file, "w") as f:
-            f.write("Total tests: " + str(len(self.passed_tests) + len(self.failed_tests)) + "\n")
-            f.write(":green_circle: " + str(len(self.passed_tests)) + " passed\n") 
-            f.write(":red_circle: " + str(len(self.failed_tests)) + " failed\n")
+            self.add_component_version_table(f)
+
+            f.write("Total tests: " + str(len(self.tests)) + "\n")
+            f.write(":green_circle: " + str(self.passed_tests) + " passed\n") 
+            f.write(":red_circle: " + str(self.failed_tests) + " failed\n")
```

Comment:
According to the [documentation](https://robot-framework.readthedocs.io/en/stable/autodoc/robot.result.html#module-robot.result.visitor) the statistics can be loaded like this:

```suggestion
            stats = result.statistics

            self.add_component_version_table(f)

            f.write(f"Total tests: {stats.tags.combined[0].total}\n")
            f.write(f":green_circle: {stats.total.passed} passed\n") 
            f.write(f":red_circle: {stats.total.failed} failed\n")
```

Reply:
This part in the `visit_test` method could be removed.
```py
if test.status == 'FAIL':
    self.failed_tests += 1
elif test.status == 'PASS':
    self.passed_tests += 1
```